mender-update/daemon/states.cpp

```diff
@@ -399,6 +483,75 @@ void UpdateCleanupState::OnEnterUpdateState(Context &ctx, sm::EventPoster<StateE
 			}));
 }
 
+void ClearArtifactDataState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
+	auto err = ctx.mender_context.GetMenderStoreDB().WriteTransaction([](kv_db::Transaction &txn) {
+		// Remove state data, since we're done now.
+		auto err = txn.Remove(main_context::MenderContext::state_data_key);
+		if (err != error::NoError) {
+			return err;
+		}
+		return txn.Remove(main_context::MenderContext::state_data_key_uncommitted);
+	});
+	if (err != error::NoError) {
+		log::Error("Error removing artifact data: " + err.String());
+		poster.PostEvent(StateEvent::Failure);
+		return;
+	}
+
+	poster.PostEvent(StateEvent::Success);
+}
+
+void StateLoopState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
+	assert(ctx.deployment.state_data);
+	auto &artifact = ctx.deployment.state_data->update_info.artifact;
+
+	// Mark update as inconsistent.
+	artifact.artifact_name = AddInconsistentSuffix(artifact.artifact_name);
```

Comment:
Q: Why not make a new string?

Reply:
We are..? I'm not sure I understand you.

Reply:
I meant not store it back in the artifact structure

Reply:
Ah, now I see what you mean. Right, because it's hiding the original name... Yes, I think you're right, it's better to keep the original information and make a new string. I'll change it.

Reply:
Handled in https://github.com/mendersoftware/mender/pull/1331.